File: Windows - console/Lm1LongCommand.cpp
#include "Lm1LongCommand.h"

const int AMOUNT_OF_COMMANDS(22);
const int MAX_SIZE_OF_REGISTER(65535);
const int CMD_LIST[AMOUNT_OF_COMMANDS] = { 0, 16, 1, 2, 3, 19, 4, 20, 5, 6, 7,
	128, 129, 130, 131, 132, 133, 134, 147, 148, 149, 150 };

Lm1LongCommand::Lm1LongCommand(){
	command = std::vector<int>(4, 0);
}

Lm1LongCommand::Lm1LongCommand(int cmd, int r_1, int m_2, int a_2){
	this->command = { cmd, r_1, m_2, a_2 };

	if ((r_1 < 0 || r_1 > MAX_SIZE_OF_REGISTER) ||
		(m_2 < 0 || m_2 > MAX_SIZE_OF_REGISTER) ||
		(a_2 < 0 || a_2 > MAX_SIZE_OF_REGISTER))
		throw std::out_of_range("Memory bounds violation");

	for (int i = 0; i < AMOUNT_OF_COMMANDS; i++)
		if (cmd == CMD_LIST[i])
			return;

	throw std::invalid_argument("Invalid command!");
}

std::vector<int>& Lm1LongCommand::get(){
	return this->command;
}

int Lm1LongCommand::get_value()const{
	return this->command.front();
}

File: Windows - GUI CS/LM_RM/LM1LongCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LM_GUI_UP
{
    class LM1LongCommand:Command
    {
        public LM1LongCommand(int cmd, int r1, int m2, int a2)
        {
            CMD_LIST = new int[]{ 0, 16, 1, 2, 3, 19, 4, 20, 5, 6, 7,
	        128, 129, 130, 131, 132, 133, 134, 147, 148, 149, 150, 300 };

            command = new List<int>() { cmd, r1, m2, a2 };

            if ((r1 < 0 || r1 > MAX_SIZE_OF_REGISTER) ||
                (m2 < 0 || m2 > MAX_SIZE_OF_REGISTER) ||
                (a2 < 0 || a2 > MAX_SIZE_OF_REGISTER))
                throw new Exception("runtime error: Register bounds violation.");

            for (int i = 0; i < CMD_LIST.Length; i++)
                if (cmd == CMD_LIST[i])
                    return;

            throw new Exception("runtime error: Invalid command number.");
        }

        public override string ToString()
        {
            return TextFormat.addZeros(Convert.ToString(command[0], 16).ToUpper(), 2)
                + ' ' + TextFormat.addZeros(Convert.ToString(command[1], 16).ToUpper(), 2)
                + ' ' + TextFormat.addZeros(Convert.ToString(command[2], 16).ToUpper(), 2)
                + ' ' + TextFormat.addZeros(Convert.ToString(command[3], 16).ToUpper(), 4);
        }
    }
}
